docs/stm32/general.rst

```diff
@@ -0,0 +1,36 @@
+.. _stm32_general:
+
+General Information about the STM32 port
+========================================
+
+STM32 is a family of microcontroller ICs based on 32-bit RISC architecture
+such as Cortex-M33F, Cortex-M7F, Cortex-M4F, Cortex-M3, Cortex-M0+, and
+Cortex-M0 cores.
+
+The STM microcontroller boards provides the facility of accessing the memory
+directly, to perform any read or write function.
+
+It exposes three objects::
+
+* stm.mem8 : access 8 bits of memory
+
+* stm.mem16 : access 16 bits of memory
+
+* stm.mem32 : access a 32 bit word memory
+
+They can be used in combination with the offset constants to read, write and
+control registers of the MCU hardware peripherals and all other areas of MCU's
+memory.
+
+Example:-
+
+..code-block:: python3
+
+    stm.mem32[stm.TIM3 + stm.TIM_CR1] 
```

Comment:
I think this example should use the GPIO because that's easier and easy to see an effect.  Eg:
```
# set PA2 high
stm.mem32[stm.GPIOA + stm.GPIO_BSRR] = 1 << 2

# read PA3
value = (stm.mem32[stm.GPIOA + stm.GPIO_IDR] >> 3) & 1
```